File: camp2/gara/flagsforkeys/flagsforkeys.py
key1 = b"JIiEv3"
key2 = b"Kg9FRj"
key3 = b"xe8zh2"
map = [5, 1, 0, 2, 3, 4]

p1 = bytes([(c - 4) % 256 for c in key1])

p2 = bytes([c ^ 6 for c in key2])

p3 = bytearray(6)
for i in range(6):
    p3[map[i]] = key3[i]
p3 = bytes(p3)

key = p1 + b'-' + p2 + b'-' + p3

print(key.decode())

File: camp/gara/flagsforkeys/flagsforkeys.c
key1 = "JIiEv3";
int check_1(long p)
{
  char c [6];
  
  for (int i = 0; i < 6; i++) {
    c[i] = p[i] + '\x04';
  }
  int ret = strncmp(key_1, c, 6);

  return ret != 0;
}

key2 = "Kg9FRj";
int check_2(long p)
{
  char c [6];
  
  for (int i = 0; i < 6; i++) {
    c[i] = p[i] ^ 6;
  }
  int ret = strncmp(key_2, c, 6);

  return ret != 0;
}

key3 = "xe8zh2";
int check_3(long p1)
{
  char c [6];
  
  for (int i = 0; i < 6; i++) {
    c[i] = p1[map[i]];
  }
  int ret = strncmp(key_3, c, 6);

  return ret != 0;
}
